src/ansys/dpf/core/data_tree.py

```diff
@@ -535,18 +537,18 @@ def get_sub_tree_names(self):
         return coll_obj.get_integral_entries()
 
     def __to_dict(self, dic):
-        for attribute_name in self.get_attribute_names():
+        for attribute_name in self.get_attribute_names:
             dic[attribute_name] = self.get_as(attribute_name)
 
-        for sub_tree_name in self.get_sub_tree_names():
+        for sub_tree_name in self.get_sub_tree_names:
             sub_tree = self.get_as(sub_tree_name, types.data_tree)
             sub_dic = {}
             sub_tree.__to_dict(sub_dic)
             dic[sub_tree_name] = sub_dic
 
     def to_dict(self):
         """
-        Returns a dictionary representation of the DataTree
```

Comment:
How do you enforce the fact it is read-only if you return an actual Python `dict`?

Reply:
It was originally a comment from @cbellot000 meaning that modifying the attributes & attribute-names in the dict didn't actually modify anything the DataTree. Which on second thought, the fact that we return a dict makes it clear enough. But no harm in leaving there I think.